include/reactor-uc/macros.h

```diff
@@ -310,54 +339,80 @@
+    Port *downstreams[(DownstreamSize)];                                                                               \
+  } ParentName##_##ConnName;
 
-#define DEFINE_LOGICAL_CONNECTION_CTOR(ParentName, ReactorName, OutputPort, DownstreamSize)                            \
-  void ParentName##_##ReactorName##_conn_##OutputPort##_ctor(ParentName##_##ReactorName##_conn_##OutputPort *self,     \
-                                                             Reactor *parent) {                                        \
-    LogicalConnection_ctor(&self->super, parent, (Port **)self->downstreams,                                           \
+#define DEFINE_LOGICAL_CONNECTION_CTOR(ParentName, ConnName, DownstreamSize)                                           \
+  void ParentName##_##ConnName##_ctor(ParentName##_##ConnName *self, Reactor *parent) {                                \
+    LogicalConnection_ctor(&self->super, parent, self->downstreams,                                                    \
                            sizeof(self->downstreams) / sizeof(self->downstreams[0]));                                  \
   }
 
-#define LOGICAL_CONNECTION_INSTANCE(ParentName, ReactorName, OutputPort)                                               \
-  ParentName##_##ReactorName##_conn_##OutputPort conn_##OutputPort;
-#define CONTAINED_OUTPUT_CONNECTIONS(ReactorName, OutputPort, NumConnsOut)                                             \
-  Connection *_conns_##ReactorName##_##OutputPort##_out[NumConnsOut];
+#define LOGICAL_CONNECTION_INSTANCE(ParentName, ConnName) ParentName##_##ConnName ConnName;
 
-#define INITIALIZE_LOGICAL_CONNECTION(ParentName, ReactorName, OutputPort)                                             \
-  ParentName##_##ReactorName##_conn_##OutputPort##_ctor(&self->conn_##OutputPort, &self->super)
-
-#define DEFINE_DELAYED_CONNECTION_STRUCT(ParentName, ReactorName, OutputPort, DownstreamSize, BufferType, BufferSize,  \
-                                         Delay)                                                                        \
+#define CONTAINED_OUTPUT_CONNECTIONS(ReactorName, OutputPort, NumConnsOut)                                             \
+  Connection *_conns_##ReactorName##_##OutputPort[NumConnsOut];
+
+#define CONTAINED_INPUT_SOURCES(ReactorName, InputPort, NumSources)                                                    \
+  Reaction *_sources_##ReactorName##_##InputPort[NumSources];
+
+#define CONTAINED_OUTPUT_EFFECTS(ReactorName, OutputPort, NumEffects)                                                  \
+  Reaction *_effects_##ReactorName##_##OutputPort[NumEffects];
+
+#define CONTAINED_OUTPUT_OBSERVERS(ReactorName, OutputPort, NumObservers)                                              \
+  Reaction *_observers_##ReactorName##_##OutputPort[NumObservers];
+
+#define DEFINE_CONTAINED_OUTPUT_ARGS(ReactorName, OutputPort)                                                          \
```

Comment:
This macro creates an OutputExternalCtorArgs object and initializes its fields so that we can later pass it to the ctor of the child